djstripe/models.py

```diff
@@ -74,34 +56,44 @@ def __str__(self):
         return "<{message}, pk={pk}, Event={event}>".format(message=self.message, pk=self.pk, event=self.event)
 
 
-@python_2_unicode_compatible
-class Event(StripeObject):
+class Event(StripeEvent):
+    objects = models.Manager()
+    customer = models.ForeignKey("Customer", null=True,
+                                 help_text="In the event that there is a related customer, this will point to that "
+                                           "Customer record")
+    valid = models.NullBooleanField(null=True,
+                                    help_text="Tri-state bool. Null == validity not yet confirmed. Otherwise, this "
+                                              "field indicates that this event was checked via stripe api and found "
+                                              "to be either authentic (valid=True) or in-authentic (possibly "
+                                              "malicious)")
 
-    kind = models.CharField(max_length=250)
-    livemode = models.BooleanField(default=False)
-    customer = models.ForeignKey("Customer", null=True)
-    webhook_message = JSONField()
-    validated_message = JSONField(null=True)
-    valid = models.NullBooleanField(null=True)
-    processed = models.BooleanField(default=False)
+    processed = models.BooleanField(default=False, help_text="If validity is performed, webhook event processor(s) "
+                                                             "may run to take further action on the event. Once these "
+                                                             "have run, this is set to True.")
 
     @property
     def message(self):
-        return self.validated_message
-
-    def __str__(self):
-        return "<{kind}, stripe_id={stripe_id}>".format(kind=self.kind, stripe_id=self.stripe_id)
+        if self.valid:
+            return self.webhook_message
 
     def validate(self):
-        evt = stripe.Event.retrieve(self.stripe_id)
-        self.validated_message = json.loads(
+        """
+        The original contents of the Event message comes from a POST to the webhook endpoint. This data
+        must be confirmed by re-fetching it and comparing the fetched data with the original data. That's what
+        this function does.
+
+        This function makes an API call to Stripe to re-download the Event data. It then
+        marks this record's valid flag to True or False.
+        """
+        evt = self.api_retrieve()
+        validated_message = json.loads(
             json.dumps(
                 evt.to_dict(),
                 sort_keys=True,
                 cls=stripe.StripeObjectEncoder
```

Comment:
I think this can just be removed:
```python
class StripeObjectEncoder(util.json.JSONEncoder):

    def __init__(self, *args, **kwargs):
        warnings.warn(
            '`StripeObjectEncoder` is deprecated and will be removed in '
            'version 2.0 of the Stripe bindings.  StripeObject is now a '
            'subclass of `dict` and is handled natively by the built-in '
            'json library.',
            DeprecationWarning)
        super(StripeObjectEncoder, self).__init__(*args, **kwargs)
```